### DISCUSSION TITLE
out of dialog INFO

### DISCUSSION THREAD

**Discussion by MaJiangZh** [resiprocate/resiprocate#343] [Q&A] [answered]
Hi,
  How can i modify resiprocate-1.12 source code and make it safely support out of dialog INFO message? In my scenario, I need to send an INFO message with a body, and the protocol stack will call back out it after receiving the INFO message.Here are my modifications. I have recalled the INFO message to onMessageArrived:
![image](https://github.com/resiprocate/resiprocate/assets/16238193/aefe36f3-d1ac-47a1-8100-572222bf9783)
May I ask if there is a problem with my modifications?Thanks a lot.

**Comment by sgodin** [comment, marked as answer]
Your modifications look sane to me, however it's hard to really tell since
your version of DialogSet is so different from resip main.  Your line
numbers are off by thousands... perhaps you have other customizations that
are affecting things?

I wouldn't use an out of dialog MESSAGE request to model after though,
since PagerMessages try to track some state.

In DialogSet::dispatch - if the message is an INFO request you probably
want to end up calling makeServerOutOfDialog, and if it's an INFO response
(ie: you sent the request and are now getting your first response) you
probably want to end up calling makeClientOutOfDialogReq.  You will be
following the same processing as an OOD OPTIONS message.

https://github.com/resiprocate/resiprocate/wiki/Handling-Options-Requests

--
Scott Godin
SIP Spectrum, Inc.
www.sipspectrum.com
